### PR TITLE
Update Cloud SQL sample apps with Kubernetes Engine yaml files.

### PR DESCRIPTION
**Pull request by jsimonweb**

### CODE REVIEW

File: cloud-sql/mysql/servlet/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+spec:
+  selector:
+    matchLabels:
+      app: gke-cloud-sql-app
+  template:
+    metadata:
+      labels:
+        app: gke-cloud-sql-app
+    spec:
+      serviceAccountName: <YOUR-KSA-NAME>
```
Comment (lesv) [inline cloud-sql/mysql/servlet/deployment.yaml:29]: I'm going to know what KSA-NAME is?

Comment (lesv) [reply cloud-sql/mysql/servlet/deployment.yaml:29]: When I google it, I don't get what I expect.  Perhaps a link to the docs?

File: cloud-sql/mysql/servlet/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        - name: INSTANCE_CONNECTION_NAME
+          value: <INSTANCE_CONNECTION_NAME>
+        - name: DB_HOST
+          value: "127.0.0.1"
+        - name: DB_PORT
+          value: "3306"  
+        - name: DB_USER
+          valueFrom:
+            secretKeyRef:
+              name: <YOUR-DB-SECRET>
```
Comment (lesv) [inline cloud-sql/mysql/servlet/deployment.yaml:50]: Is it really best practices to put these in a YAML?

File: cloud-sql/mysql/servlet/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+              key: username
+        - name: DB_PASS
+          valueFrom:
+            secretKeyRef:
+              name: <YOUR-DB-SECRET>
```
Comment (lesv) [inline cloud-sql/mysql/servlet/deployment.yaml:55]: really - shouldn't secret manager somehow be involved or mounted secrets?

File: cloud-sql/mysql/servlet/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+              key: password
+        - name: DB_NAME
+          valueFrom:
+            secretKeyRef:
+              name: <YOUR-DB-SECRET>
```
Comment (lesv) [inline cloud-sql/mysql/servlet/deployment.yaml:60]: again - best practices?

File: cloud-sql/mysql/servlet/service-account.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START cloud_sql_mysql_java_gke_quickstart_sa]
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: <YOUR-KSA-NAME> # TODO(developer): replace this value.
```
Comment (lesv) [inline cloud-sql/mysql/servlet/service-account.yaml:19]: Try searching for `KSA Name` - tell me if you think that's helpful.

File: cloud-sql/postgres/servlet/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+spec:
+  selector:
+    matchLabels:
+      app: gke-cloud-sql-app
+  template:
+    metadata:
+      labels:
+        app: gke-cloud-sql-app
+    spec:
+      serviceAccountName: <YOUR-KSA-NAME>
```
Comment (lesv) [inline cloud-sql/postgres/servlet/deployment.yaml:29]: ditto this file

**Review by lesv — APPROVED** [review]
I don't like the use of KSA Name, either link to docs, or explain that it is created later in the sample.

I don't like hard coded secrets.  I'd prefer you use mounted secrets, or explain that in production that they might prefer to.

File: cloud-sql/mysql/servlet/deployment.yaml (hunk 2 above)
Comment (kurtisvg) [reply cloud-sql/mysql/servlet/deployment.yaml:50]: This isn't the value of the secret, it's a reference to the name of the secret object in K8s.

File: cloud-sql/mysql/servlet/service-account.yaml (hunk 5 above)
Comment (kurtisvg) [reply cloud-sql/mysql/servlet/service-account.yaml:19]: KSA stands for "Kubernetes Service Account", which will be explained in the quickstart that links to this documentation. Jonathan is working on the quickstart as we speak, and will update the README to the link when complete.

**Review by kurtisvg — APPROVED** [review]

**Pull request merged**